lib/api/api-fetch/response.js

```diff
@@ -1,141 +1,367 @@
+    // TODO: isomorphic encoded?
+    const value = parsedURL.toString()
+
+    // 7. Append `Location`/value to responseObject’s response’s header list.
+    responseObject[kState].headersList.push('location', value)
+
+    // 8. Return responseObject.
+    return responseObject
+  }
+
+  // https://fetch.spec.whatwg.org/#dom-response
+  constructor (body = null, init = {}) {
+    // 1. If init["status"] is not in the range 200 to 599, inclusive, then
+    // throw a RangeError.
+    if ('status' in init) {
+      if (!Number.isFinite(init.status)) {
+        throw new TypeError()
+      }
+
+      if (init.status < 200 || init.status > 599) {
+        throw new RangeError()
+      }
+    }
+
+    if ('statusText' in init) {
+      if (typeof init.statusText !== 'string') {
+        throw new TypeError()
+      }
+
+      // 2. If init["statusText"] does not match the reason-phrase token
+      // production, then throw a TypeError.
+      // See, https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.2:
+      //   reason-phrase  = *( HTAB / SP / VCHAR / obs-text )
+      // TODO
+    }
+
+    // 3. Set this’s response to a new response.
+    this[kState] = makeResponse({})
+
+    // 4. Set this’s headers to a new Headers object with this’s relevant
+    // Realm, whose header list is this’s response’s header list and guard
+    // is "response".
+    // TODO: relevant Realm?
+    this[kHeaders] = new Headers()
+    this[kHeaders][kGuard] = 'response'
+    this[kHeaders][kHeadersList] = this[kState].headersList
+
+    // 5. Set this’s response’s status to init["status"].
+    if ('status' in init) {
+      this[kState].status = init.status
+    }
+
+    // 6. Set this’s response’s status message to init["statusText"].
+    if ('statusText' in init) {
+      this[kState].statusText = init.statusText
+    }
+
+    // 7. If init["headers"] exists, then fill this’s headers with init["headers"].
+    if ('headers' in init) {
+      fill(this[kState].headersList, init.headers)
+    }
+
+    // 8. If body is non-null, then:
+    if (body !== null) {
+      // 1. If init["status"] is a null body status, then throw a TypeError.
+      if (nullBodyStatus.includes(init.status)) {
+        throw new TypeError()
+      }
+
+      // 2. Let Content-Type be null.
+      // 3. Set this’s response’s body and Content-Type to the result of
+      // extracting body.
+      const [extractedBody, contentType] = extractBody(body)
+      this[kState].body = extractedBody
+
+      // 4. If Content-Type is non-null and this’s response’s header list does
+      //  not contain `Content-Type`, then append `Content-Type`/Content-Type
+      // to this’s response’s header list.
+      if (contentType && !this.headers.has('content-type')) {
+        this.headers.set('content-type', contentType)
+      }
     }
   }
 
+  get [Symbol.toStringTag] () {
+    return this.constructor.name
+  }
+
+  toString () {
+    return Object.prototype.toString.call(this)
+  }
```

Comment:
Is this necessary as well?

Reply:
Don't know. Grabbed it from node-fetch. wdyt?

Reply:
I would remove from now